mautic/mautic
# Fix unique vs returning visitors widget

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
Unique vs returning visitors dashboard widget return wrong data. This PR try rebuild queryto return expected data

Query before

`SELECT COUNT(t.id) AS count FROM page_hits t WHERE t.date_hit BETWEEN :dateFrom AND :dateTo`

`SELECT COUNT(t.id) AS count FROM page_hits t WHERE (lead_id IS NULL) AND (t.date_hit BETWEEN :dateFrom AND :dateTo)`

Query after

- `SELECT COUNT(*) as count FROM (SELECT t.lead_id FROM page_hits t WHERE (lead_id IS NOT NULL) AND (date_left IS NULL) AND (redirect_id IS NULL) AND (email_id IS NULL) AND (t.date_hit BETWEEN :dateFrom AND :dateTo) GROUP BY t.lead_id) tt`
- `SELECT COUNT(*) as count FROM (SELECT t.lead_id FROM page_hits t WHERE (lead_id IS NOT NULL) AND (date_left IS NULL) AND (redirect_id IS NULL) AND (email_id IS NULL) AND (t.date_hit BETWEEN :dateFrom AND :dateTo) GROUP BY t.lead_id HAVING COUNT(t.id) > 1) tt`




#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create unique vs returning visitors dashboard widget on your instances with page_hits data
3. Check If you see correct pie graph
